local-tests/setup/tinny-operations.ts

```diff
@@ -0,0 +1,209 @@
+ * It parses the process arguments to find flags that specify filters, typically used to limit test execution to specific tests.
+ *
+ * @returns {string[]} An array of filter strings extracted from the command line arguments.
+ *
+ * @example
+ * // Assume the command line is: node script.js --filter=test1,test2
+ */
+export const getFiltersFlag = (): string[] => {
+  const filterArg = process.argv.find((arg) => arg.startsWith('--filter='));
+  return filterArg ? filterArg.replace('--filter=', '').split(',') : [];
+};
+
+/**
+ * Runs the tests in the provided `tests` object in a synchronous manner.
+ * Each test is executed in a loop with a maximum number of attempts specified by `devEnv.processEnvs.MAX_ATTEMPTS`.
+ * Skipped, failed, and passed tests are tracked and logged.
+ * At the end, a test report is printed with the number of passed, failed, and skipped tests.
+ * If any test fails, the process exits with an error code. Otherwise, it exits successfully.
+ *
+ * @param {Object} options - The options object.
+ * @param {Object} options.tests - The tests to run.
+ * @param {TinnyEnvironment} options.devEnv - The development environment.
+ * @returns {Promise<void>} - A promise that resolves when all tests have been executed.
+ */
+export const runInBand = async ({
+  tests,
+  devEnv,
+}: {
+  tests: any;
+  devEnv: TinnyEnvironment;
+}): Promise<void> => {
+  const filters = getFiltersFlag();
+  const testsToRun = Object.entries(tests).filter(
+    ([testName]) => filters.length === 0 || filters.includes(testName)
+  );
+
+  // Initialize arrays to keep track of skipped, failed, and passed tests
+  let skippedTests: string[] = [];
+  let failedTests: string[] = [];
+  let passedTests: string[] = [];
+
+  // Iterate over each test and run it in series
+  for (const [testName, testFunction] of testsToRun) {
+    const maxAttempts = devEnv.processEnvs.MAX_ATTEMPTS;
+    let attempts = 0;
+    let testPassed = false;
+
+    while (attempts < maxAttempts && !testPassed) {
+      const startTime = performance.now();
+
+      try {
+        console.log(`Attempt ${attempts + 1} for ${testName}...`);
+
+        // @ts-ignore
+        await testFunction(devEnv);
+        testPassed = true;
+
+        const endTime = performance.now();
+        const timeTaken = (endTime - startTime).toFixed(2);
+        console.log(`${testName} - Passed (${timeTaken} ms)`);
+        passedTests.push(`${testName} (Passed in ${timeTaken} ms)`);
+      } catch (error) {
+        if (error.message === 'LIT_IGNORE_TEST') {
+          skippedTests.push(`${testName} (Skipped)`);
+          break;
+        }
+        attempts++;
+        if (attempts >= maxAttempts) {
+          const endTime = performance.now();
+          const timeTaken = (endTime - startTime).toFixed(2);
+          console.error(
+            `${testName} - Failed after ${maxAttempts} attempts (${timeTaken} ms)`
+          );
+          console.error(`Error: ${error}`);
```

Comment:
Would be very valuable to print the stack trace from the failed test's error if it has one, instead of just the stringified error